lib/events.js

```diff
@@ -0,0 +1,132 @@
+var EventEmitter = require('eventemitter3'),
+    Inheritable = require('./inheritable'),
+    Events, callEventEmitter;
+
+/**
+ * bind events on attributes or model
+ * @param  {EventEmitter} eventEmitter
+ * @param  {string}   method of EventEmitter
+ * @param  {string}   [attribute] space separated list
+ * @param  {string}   event space separated list
+ * @param  {Function} cb
+ * @param  {*}   [ctx]
+ */
+callEventEmitter = function (eventEmitter, method, attribute, event, cb, ctx) {
+    if (typeof event !== 'string') {
+        ctx = cb;
+        cb = event;
+        event = attribute;
+        attribute = undefined;
+    }
+    ctx = ctx || this;
+    if (attribute) {
+        attribute.split(/\s+/).forEach(function (attribute) {
+            event.split(/\s+/).forEach(function (event) {
+                eventEmitter[method](event + ':' + attribute, cb, ctx);
+            });
+        });
+    } else {
+        event.split(/\s+/).forEach(function (event) {
+            eventEmitter[method](event, cb, ctx);
+        });
+    }
+};
+
+/**
+ * @class Events
+ * @extends Inheritable
+ */
+Events = Inheritable.inherit(/** @lends Events.prototype */{
+
+    __constructor: function () {
+        this._eventEmitter = new EventEmitter();
+    },
+
+    /**
+     * unbind events on attributes or model
+     * @param  {string}   [attribute] space separated list
+     * @param  {string}   event space separated list
+     * @param  {Function} cb
+     * @param  {*}   [ctx]
+     * @returns {Events}
+     */
+    un: function (attribute, event, cb, ctx) {
+        this._callEventEmitter('removeListener', attribute, event, cb, ctx);
+        return this;
+    },
+
+    /**
+     * bind events on attributes or model
+     * @param  {string}   [attribute] space separated list
+     * @param  {string}   event space separated list
+     * @param  {Function} cb
+     * @param  {*}   [ctx]
+     * @returns {Events}
+     */
+    on: function (attribute, event, cb, ctx) {
+        this._callEventEmitter('on', attribute, event, cb, ctx);
+        return this;
+    },
+
+    /**
+     * @see EventEmitter.emit
+     * @returns {Events}
+     */
+    trigger: function () {
+        var callEmitter = this.__self._eventEmitter,
+            allEventArgs;
+        this._eventEmitter.emit.apply(this._eventEmitter, arguments);
+        if (callEmitter) {
+            callEmitter.emit.apply(callEmitter, arguments);
+        }
+        // trigger `all` event
+        allEventArgs = Array.prototype.slice.call(arguments, 0);
+        allEventArgs.unshift('all');
```

Comment:
Why is `all` event triggered with string `all` first argument? All other events doesn't.